x/feegrant/keeper/keeper.go

```diff
@@ -14,17 +15,19 @@ import (
 // Keeper manages state of all fee grants, as well as calculating approval.
 // It must have a codec with all available allowances registered.
 type Keeper struct {
-	cdc        codec.BinaryMarshaler
-	storeKey   sdk.StoreKey
-	authKeeper types.AccountKeeper
+	cdc          codec.BinaryMarshaler
+	storeKey     sdk.StoreKey
+	authKeeper   types.AccountKeeper
+	msgSvcRouter *baseapp.MsgServiceRouter
 }
 
 // NewKeeper creates a fee grant Keeper
-func NewKeeper(cdc codec.BinaryMarshaler, storeKey sdk.StoreKey, ak types.AccountKeeper) Keeper {
+func NewKeeper(cdc codec.BinaryMarshaler, storeKey sdk.StoreKey, ak types.AccountKeeper, router *baseapp.MsgServiceRouter) Keeper {
```

Comment:
```suggestion
func NewKeeper(cdc codec.BinaryMarshaler, storeKey sdk.StoreKey, ak types.AccountKeeper) Keeper {
```

I don't think we need to add this complexity just to validate that these are valid method names.